bytecode/flags.py

```diff
@@ -0,0 +1,297 @@
+# alias to keep the 'bytecode' variable free
+from enum import IntEnum
+from collections import defaultdict
+
+import bytecode as _bytecode
+
+
+class CoFlags(IntEnum):
+    """Possible values of the co_flags attribute of Code object.
+
+    Notes
+    -----
+    We do not rely on inspect values here as some of them are missing and
+    furthermore would be version dependent.
+
+    """
+    CO_OPTIMIZED             = 0x00001  # noqa
+    CO_NEWLOCALS             = 0x00002  # noqa
+    CO_VARARGS               = 0x00004  # noqa
+    CO_VARKEYWORDS           = 0x00008  # noqa
+    CO_NESTED                = 0x00010  # noqa
+    CO_GENERATOR             = 0x00020  # noqa
+    CO_NOFREE                = 0x00040  # noqa
+    # New in Python 3.5
+    CO_COROUTINE             = 0x00080  # noqa
+    CO_ITERABLE_COROUTINE    = 0x00100  # noqa
+    # New in Python 3.6
+    CO_ASYNC_GENERATOR       = 0x00200  # noqa
+
+    # __future__ flags
+    CO_FUTURE_GENERATOR_STOP = 0x80000  # noqa
+
+
+_CAN_DEDUCE_FROM_CODE = ('optimized', 'generator', 'nofree', 'coroutine')
+
+
+class Flags:
+    """Object representing the flags set for the bytecode.
+
+    Flags can be bools indicating their state, or for some of them None
+    indicating that the value should be inferred from the ConcreteBytecode
+    object passed when converting to an integer. If no bytecode is provided the
+    default values are used.
+
+    The flags whose natural values can be inferred are:
+    - optimized
+    - generator
+    - nofree
+    - coroutine
+
+    """
+
+    def __init__(self, int_or_flags=0):
+
+        self._forced = {}
+
+        if isinstance(int_or_flags, Flags):
+            self._defaults = int_or_flags._defaults.copy()
+            self._forced = int_or_flags._forced.copy()
+        elif isinstance(int_or_flags, int):
+            self._defaults = {name.split('_', 1)[1].lower():
+                              bool(int_or_flags & val)
+                              for name, val in CoFlags.__members__.items()}
+        else:
+            msg = ("Flags object should be passed either None, an int  or a "
+                   "Flags instance at init, got {}.")
+            raise TypeError(msg.format(int_or_flags))
+
+    def __eq__(self, other):
+        if isinstance(other, int):
+            return self.to_int() == other
```

Comment:
Hum, I would prefer to cast int to Flags, rather than the opposite. Would it work?